### PR TITLE
Allow to define module front controllers layout

### PR DESCRIPTION
**Pull request by tonyyb**
Re-open : https://github.com/PrestaShop/PrestaShop/pull/7771 with good convention name.

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch       | 1.7.x
| Description?  | Currently it is not possible to define the layout of a module front controller.
| Type?         | improvement
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | -
| How to test?  | Try to define the layout of a module from AdminThemes back office controller.

### CODE REVIEW

**Comment by xBorderie** [comment]
👍

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1308,6 +1308,9 @@ protected function getThemeDir()
     public function getLayout()
     {
         $entity = $this->php_self;
+        if (empty($entity) && $this->controller_type === 'modulefront' && !empty($this->page_name)) {
+            $entity = $this->page_name;
+        }
```
Comment (Quetzacoalt91) [inline classes/controller/FrontController.php]: Hello @tonyyb,

What do you think if we just use $this->getPageName()? 
```php
$entity = $this->php_self;
if (empty($entity)) {
    $entity = $this->getPageName();
}
```

It seems to handle even more cases:
```php
    public function getPageName()
    {
        // Are we in a payment module
        $module_name = '';
        if (Validate::isModuleName(Tools::getValue('module'))) {
            $module_name = Tools::getValue('module');
        }

        if (!empty($this->page_name)) {
            $page_name = $this->page_name;
        } elseif (!empty($this->php_self)) {
            $page_name = $this->php_self;
        } elseif (Tools::getValue('fc') == 'module' && $module_name != '' && (Module::getInstanceByName($module_name) instanceof PaymentModule)) {
            $page_name = 'module-payment-submit';
        } elseif (preg_match('#^'.preg_quote($this->context->shop->physical_uri, '#').'modules/([a-zA-Z0-9_-]+?)/(.*)$#', $_SERVER['REQUEST_URI'], $m)) {
            // @retrocompatibility Are we in a module ?
            $page_name = 'module-'.$m[1].'-'.str_replace(array('.php', '/'), array('', '-'), $m[2]);
        } else {
            $page_name = Dispatcher::getInstance()->getController();
            $page_name = (preg_match('/^[0-9]/', $page_name) ? 'page_'.$page_name : $page_name);
        }

        return $page_name;
    }
```

**Comment by tonyyb** [comment]
Hi @xBorderie @Quetzacoalt91 : I think that's a good idea ! :+1:

**Review by Quetzacoalt91 — APPROVED** [review]

**Comment by xBorderie** [comment]
Thank you!

Could you try and [squash those commits into just one](http://blog.steveklabnik.com/posts/2012-11-08-how-to-squash-commits-in-a-github-pull-request)? [Another explanation of squashing](https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/).

**Comment by Quetzacoalt91** [comment]
This is okay to keep the history like this @xBorderie, that's why I merge it. ;)

Thanks @tonyyb.

**Comment by xBorderie** [comment]
Fine by me :)

**Pull request merged**